### DISCUSSION TITLE
Elaborate but safe way to start motor

### DISCUSSION THREAD

**Discussion by Shonkonray** [EdgeTX/edgetx#3801] [Show and tell]
Hello,

i wanted to share with you my motor start logic which i am using since the very beginning years ago and it saved me many times from accidents.

The idea was to have the motor only started while throttle stick is in a certain position (model dependent) and a dedicated throttle cut/start switch.
So far nothing new, but i wanted it to be a step safer than this. Because there is one flaw with this simple setup: the order in which the requirements for a successful start are cleared is not entirely defined.
Additionally, i wanted to give myself enough time to abort an unwanted start in case those two conditions are met by accident.

### In a nutshell what i wanted to have:
1) A clearly defined order in which the motor has to be started
   -> i decided that first the throttle stick has to be in position and then and only then can the start switch be activated
2) Prevent an unwanted start as good as possible, but with no hindrance to usability
3) Give me time to react in case of an unwanted start
4) Give me a clear signal that something is going to happen

Please see the screenshot of my logic (one from a plane and one from a heli) to see how this looks like.
At first it seems overly unwieldly or intimidating (and it may be), but with this i enjoy every single start. 😄

Following a short explanation of what each of those logic switches is doing and meant for.

### Logical Switch Explanation:
**L01** -> Desired throttle stick position for engine start --> a short delay ensures that its not a fluke

**L02** -> Rising edge for engine start only possible with L01 = TRUE --> this does solve my above-mentioned safety point and ensures a fixed order of how to start. If you flick the switch before the stick is in position, you have to do it again after you have brought the stick in position

**L03** -> Engine start warning sound until L04 = TRUE --> (in my case 2s, see L04 delay) this checks point 3 and 4 of my list above, and this is also what saved me many times. While working on my models the throttle stick would often be in the right position and i would lean over my TX or move it around and accidently flip my throttle cut SF. But due to the 2s delay and the warning sound i got aware of it in time and could prevent a start. I dont know why, but it is also helping me on the field to make up my mind or abort if i feel something is amiss!

**L04** -> Engine start / Flight timer start until SF is switched back

![Motor_start_logic](https://github.com/EdgeTX/edgetx/assets/35926350/ada56f1d-850f-44bc-ac6e-e353ee837aff)

In case you want to use this also, please not that there a delays and duration times included, which might be overlooked at first glance.

Have fun testing and experimenting with it. 😄

**Comment by pfeerick** [comment]
Very nice, and very well explained! :) I was actually sharing my logic with someone else in another issue, so will add that here. Looking at yours, I might actually be able to optimise it a little more by combining the logic. I also hadn't really thought to add a delay to the throttle check, so will probably add that. 

![image](https://github.com/EdgeTX/edgetx/assets/5500713/344034b6-038b-4258-9390-1ebaaf66f470)

In a nutshell

L01 gives me a pulse when SF goes down [arm switch must be going from up to down]
L02 checks if the throttle is down (with a little wiggle room) [throttle down]
L03 combines those two (so true if SF is down, and the throttle is down) [safe to arm]
L04 combines throttle down and SF up [disarm]
L05 latches if L03 true, resets if L04 true [system armed]
The purpose of L01 is to make it so if the throttle is up at the time I try to arm (SF down), I must lower the throttle, AND reset the switch (back to SF up) - basically start again - before it will arm.

Bonus extras:

L06 enables if the throttle is not down (!L02) while the system is armed [in flight]
L07 enables if SF goes down while in flight [disarm in flight detected]
L10 checks if system not armed, and no telemetry
L11 checks if system is armed, and telemetry present
L06 and L07 allow me to detect an attempt to disarm in flight, and L07 triggers a beep to let me know if I pull the throttle down while in the current state, it's bye bye quad.

L10 is used to reset my flight timers when telemetry lost and system is disarmed. Basically when I pull the battery, it auto resets the timer. L11 is used to run logging only if I have telemetry and am armed.

**Reply by Shonkonray** [reply]
The other someone with the other issue, was me again 😁 

Yes you could combine L01 and L02 by using L02 in the AND switch at L01 then L03 is obsolete. Also L06 and L07 could be combined into one with SF up at the AND switch in L06.
If you would use at L04 an Edge with SF up and at the AND Switch L02 (basically the same logic as L01 and L03 combined but with SF up instead of down), you wouldnt need L06 and L07 at all. Because with this it would be ensured that you have to move down throttle first and then flick SF to disarm. It wouldnt be any less safe than now, because right now L04 would disarm without warning anyways.
I have built this solution in the screenshot below L01 to L04

If you would like to be absolutely safe from any unwanted disarm or arm you would have to put in some time delays.
I took the liberty to build this logic too, from L10 to L18. With this logic you have 2s time delay until motor starts (with warn sound) and 2s disarm delay (with warn sound). During these delays you can safely abort any unwanted action.
Arming/disarming can only be triggered with SF switch if throttle is in position already.

L17 gives out sound if wanted
L18 is the actual motor start

Btw. with logic like this i like to use the throttle stick directly and not the input value, not that a change in the input is destroying my safety programming. 😄 
![Arming_Solution](https://github.com/EdgeTX/edgetx/assets/35926350/b2781828-659d-4aec-89b6-5cb49e63cf2f)

Have a look at it, may it be useful and its probably easier to build it than to think about it. 😄

**Reply by pfeerick** [reply]
Hm... L01 to L04 is certainly more compact than what I had, but would need to go back to the L10/L18  version again. The pre-start warning isn't so critical for me... it's either plugged in or not plugged in (not that I don't see a benefit to having it), as is the requirement for locking the disarm to throttle down then switch moved. I usually disarm in flight, on the home approach, so that the throttle down is always terminal mode ;) That's the secondary reason the beep on disarm "pending" is important. The other thing I need to shoe-horn in is a hard disarm... possibly something like T5/T6 both down, for an instant kill. It's nice to be able to discuss this and think through different options. :)

**Reply by Shonkonray** [reply]
Ah i see, i dont have any experience with drones, aside from trying a friend´s, but the concept of arm/disarm/cut/terminal mode etc. are foreign to me. 😄
I dont know where the difference is or why it would have to be any different from a plane.

My personal priority is that i can cut the engine as fast as possible without thinking about it. Since i started the hobby with helis, where this is particularly important, i kept this habit of using a cut switch (flicking stick down is not the best idea in this case 😂 ) even with my plane, although throttle down stick is also stoping the engine. Unless there are any hardware reasons on a drone (flight controler?), I would probably use the same logic on them too. Engine off with throttle to zero and a complete cut via a dedicated switch.

Anyways,  i understand that everyones prioritys/interests/habits and needs are different and according to your last message i made another logic for you to look at. Its in principle your first version cleaned up and then extended with your wish of an instant kill (L28). L31 is your kill pending sound. But now its a rather boring logic, like any other too, so my vote goes to L10 to L18. :joy:

![Disarm_V3](https://github.com/EdgeTX/edgetx/assets/35926350/7d0ad771-cf10-4fb4-afd2-e89260bbde84)

Yes, you are right, its nice and interesting to see what other people are doing to get their needs met. 😄
